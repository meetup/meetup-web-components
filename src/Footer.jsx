import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import locales from 'mwp-config/locales';

import localizedLanguageMap from 'mwp-i18n/lib/localizedLanguageMap';

import AppBadges from './media/AppBadges';
import Bounds from './layout/Bounds';
import Chunk from './layout/Chunk';
import Flex from './layout/Flex';
import FlexItem from './layout/FlexItem';
import InlineBlockList from './layout/InlineBlockList';
import GridList from './layout/GridList';
import Icon from './media/Icon';
import Section from './layout/Section';
import SelectInput from './forms/SelectInput';
import Stripe from './layout/Stripe';
import withMatchMedia from './utils/components/withMatchMedia';

import { getSocialLinks } from './utils/getSocialLinks';

export const FooterCategory = props => (
	<Chunk>
		<h4 className="text--bold margin--bottom">{props.header}</h4>
		<GridList
			columns={{
				default: 2,
				medium: 1,
			}}
			className="text--secondary text--small"
			itemClassNames="footer-item"
			items={props.items}
		/>
	</Chunk>
);

export const SocialIconsList = ({ localeCode }) => {
	const socialLinks = getSocialLinks(localeCode);
	const socialIcons = [
		<a href={socialLinks.facebook}>
			<Icon shape="external-facebookboxed" size="s" />
		</a>,
		<a href={socialLinks.twitter}>
			<Icon className="margin--left" shape="external-twitter" size="s" />
		</a>,
		<a href={socialLinks.youtube}>
			<Icon className="margin--left" shape="external-youtube" size="s" />
		</a>,
		<a href={socialLinks.instagram}>
			<Icon className="margin--left" shape="external-instagram" size="s" />
		</a>,
	];

	return <InlineBlockList items={socialIcons} />;
};

export const LanguageSelectInput = props => (
	<SelectInput
		onChange={props.onChange}
		options={locales.map(language => ({
			label: localizedLanguageMap[language],
			value: language,
		}))}
		name="languagePicker"
		value={props.currentLocaleCode}
		label="Language"
		labelClassName="visibility--a11yHide"
	/>
);

export const Footer = ({
	className,
	createMeetup,
	isLight,
	isLoggedIn,
	legalCopy,
	localeCode,
	linkSets,
	media,
	onLanguageSelect,
	socialHeader,
	subFooterLinks,
	...other
}) => {
	const classNames = cx('column-item', 'column-item--shrink', className);
	const languageShort = localeCode.split('-')[0];

	return (
		<footer id="mupFooter" className={classNames} {...other}>
			<Stripe
				inverted={!isLight}
				className={cx('footerStripe-main', {
					'footerStripe-main--isLight': isLight,
				})}
			>
				<Bounds narrow className="bounds--footer">
					<Section hasSeparator>
						<Chunk>
							<a href={createMeetup.link} className="link link--white">
								{createMeetup.text}
							</a>
						</Chunk>
					</Section>
					<Section hasSeparator className="border--none">
						<Flex direction="column" switchDirection="large">
							<FlexItem className="margin--bottom">
								<Flex direction="column" switchDirection="medium">
									{linkSets.map((linkSet, i) => (
										<FlexItem key={`linkSet-${i}`}>
											<FooterCategory {...linkSet} />
										</FlexItem>
									))}
								</Flex>
							</FlexItem>
							<FlexItem shrink>
								<Flex noGutters direction="column">
									<FlexItem shrink>
										<Chunk className="align--center atMedium_align--left">
											<h4 className="text--bold margin--bottom">
												{socialHeader}
											</h4>
											<SocialIconsList localeCode={localeCode} />
										</Chunk>
									</FlexItem>
									<FlexItem>
										<Flex
											direction={
												!media.isAtMediumUp || media.isAtLargeUp
													? 'column'
													: 'row'
											}
											rowReverse={
												media.isAtMediumUp &&
												!media.isAtLargeUp &&
												!isLoggedIn
											}
											justify={
												media.isAtMediumUp &&
												!media.isAtLargeUp &&
												!isLoggedIn
													? 'spaceBetween'
													: null
											}
										>
											{!isLoggedIn && (
												<FlexItem shrink className="margin--top">
													<Chunk>
														<LanguageSelectInput
															currentLocaleCode={localeCode}
															onChange={onLanguageSelect}
														/>
													</Chunk>
												</FlexItem>
											)}
											<FlexItem shrink>
												<Chunk className="align--center atMedium_align--left margin--top">
													<AppBadges language={languageShort} />
												</Chunk>
											</FlexItem>
										</Flex>
									</FlexItem>
								</Flex>
							</FlexItem>
						</Flex>
					</Section>
				</Bounds>
			</Stripe>
			<Stripe
				inverted={!isLight}
				className={cx('footerStripe-legal border--none', {
					'footerStripe-legal--isLight': isLight,
				})}
			>
				<Bounds>
					<div className="padding--all">
						<InlineBlockList
							className="text--small align--center atMedium_align--left footerList-legal"
							separator="·"
							items={[
								<span>{`© Meetup ${new Date().getFullYear()} LLC`}</span>
							].filter(item => item)}
						/>
						<InlineBlockList
							className="text--small align--center atMedium_align--left margin--top"
							separator="·"
							items={[...subFooterLinks]}
						/>
					</div>
				</Bounds>
			</Stripe>
		</footer>
	);
};

Footer.defaultProps = {
	createMeetup: {
		text: 'Create a Meetup',
		link: '/create/',
	},
	linkSets: [
		{
			header: 'Your Account',
			items: [
				<a href="/register">Settings</a>,
				<a href="/logout">Log out</a>,
				<a href="/help">Help</a>,
			],
		},
		{
			header: 'Discover',
			items: [
				<a href="/groups">Groups</a>,
				<a href="/calendar">Calendar</a>,
				<a href="/topics">Topics</a>,
				<a href="/cities">Cities</a>,
			],
		},
		{
			header: 'Meetup',
			items: [
				<a href="/about">About</a>,
				<a href="/pro">Meetup Pro</a>,
				<a href="/jobs">Careers</a>,
				<a href="/apps">Apps</a>,
				<a href="/meetup_api">API</a>,
			],
		},
	],
	localeCode: 'en-US',
	subFooterLinks: [
		<a href="/terms">Terms of Service</a>,
		<a href="/privacy">Privacy Policy</a>,
		<a href="/cookie_policy">Cookie Policy</a>,
	],
	socialHeader: 'Follow Us',
};

Footer.propTypes = {
	/** Additional class name/s to add to the `<footer/>` element  */
	className: PropTypes.string,

	/** Takes a URL and a text label to render the link to create a new Meetup group  */
	createMeetup: PropTypes.shape({
		link: PropTypes.string.isRequired,
		text: PropTypes.string.isRequired,
	}).isRequired,

	/** Renders a light footer background instead of dark. Used for 404's and error pages  */
	isLight: PropTypes.bool,

	/** Used to decide which content to render based on whether user is logged into their Meetup account  */
	isLoggedIn: PropTypes.bool,

	/** Fine print required by Meetup's legal team */
	legalCopy: PropTypes.string,

	/** The current user's locale code, which is used to render the correct social media links and language dropdown */
	localeCode: PropTypes.string.isRequired,

	/** Collections of links to render in the footer. For example: "Your Account" links, "Discovery" links, and Meetup-specific links */
	linkSets: PropTypes.arrayOf(
		PropTypes.shape({
			header: PropTypes.string.isRequired,
			items: PropTypes.arrayOf(PropTypes.element).isRequired,
		})
	).isRequired,

	/** Function to call when a user selects a language */
	onLanguageSelect: PropTypes.func.isRequired,

	/** Label for our social media links. e.g.: "Follow us" */
	socialHeader: PropTypes.string,

	/** Small-print links. e.g.: "Terms of Servuce" */
	subFooterLinks: PropTypes.arrayOf(PropTypes.element).isRequired,
};

export default withMatchMedia(Footer);
