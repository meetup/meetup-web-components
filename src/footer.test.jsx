import React from 'react';
import { shallow, mount } from 'enzyme';

import config from 'mwp-config';

import { Footer, FooterCategory, LanguageSelectInput, SocialIconsList } from './Footer';

const DEFAULT_LANG = 'en-US';
const onLanguageSelect = jest.fn();

const footerLinkSets = [
	{
		header: 'Your Account',
		items: [
			<a href="/register">Settings</a>,
			<a href="/logout">Log out</a>,
			<a href="/help">Help</a>,
		],
	},
	{
		header: 'Discover',
		items: [
			<a href="/groups">Groups</a>,
			<a href="/calendar">Calendar</a>,
			<a href="/topics">Topics</a>,
			<a href="/cities">Cities</a>,
		],
	},
	{
		header: 'Meetup',
		items: [
			<a href="/about">About</a>,
			<a href="/pro">Meetup Pro</a>,
			<a href="/jobs">Careers</a>,
			<a href="/apps">Apps</a>,
			<a href="/blog">Blog</a>,
		],
	},
];

const subFooterLinks = [
	<a href="/terms">Terms</a>,
	<a href="/privacy">Privacy</a>,
	<a href="/cookie_policy">Cookies</a>,
];

const MOCK_DEFAULT_STATE = {
	legalCopy: 'Mock legal copy',
	localeCode: 'en-US',
	linkSets: footerLinkSets,
	onLanguageSelect: onLanguageSelect,
	subFooterLinks: subFooterLinks,
	createMeetup: {
		text: 'Create a Meetup',
		link: '/create/',
	},
	media: { isAtLargeUp: false },
};

const MOCK_AUTHENTICATED_STATE = { isLoggedIn: true };

describe('Footer', () => {
	const footerComponent = shallow(<Footer {...MOCK_DEFAULT_STATE} />);
	it('exists', () => {
		expect(footerComponent).toMatchSnapshot();
	});
	it('should render social media icon list', () => {
		const socialMediaIconList = footerComponent.find(SocialIconsList);
		expect(socialMediaIconList.exists()).toBe(true);
	});
	it('should render FooterCategory with linkSet', () => {
		const footerCategories = shallow(<FooterCategory {...footerLinkSets[0]} />);
		expect(footerCategories).toMatchSnapshot();
	});
});

describe('Footer unauthenticated user', () => {
	const footerComponent = shallow(<Footer {...MOCK_DEFAULT_STATE} />);
	it('shows language picker', () => {
		const langSelect = footerComponent.find(LanguageSelectInput);

		expect(langSelect.length).toBeGreaterThan(0);
	});
});

describe('Footer authenticated user', () => {
	const footerComponent = shallow(
		<Footer isLoggedIn {...MOCK_DEFAULT_STATE} {...MOCK_AUTHENTICATED_STATE} />
	);
	it('does *not* show language picker', () => {
		const langSelect = footerComponent.find(LanguageSelectInput);
		expect(langSelect.length).toBe(0);
	});
});

describe('LanguageSelectInput', () => {
	it('exists', () => {
		const langSelectInput = shallow(<LanguageSelectInput language={DEFAULT_LANG} />);
		expect(langSelectInput).toMatchSnapshot();
	});
	it('should render the correct language value', () => {
		const langSelectInput = mount(<LanguageSelectInput language={DEFAULT_LANG} />);
		const selectWrapper = langSelectInput.find('select');
		expect(selectWrapper.prop('value')).toEqual(DEFAULT_LANG);
	});
	it('should call onLanguageSelect', () => {
		const footerWrapper = mount(<Footer {...MOCK_DEFAULT_STATE} />);
		const selectWrapper = footerWrapper.find(LanguageSelectInput).find('select');
		expect(onLanguageSelect).not.toHaveBeenCalled();
		selectWrapper.simulate('change');
		expect(onLanguageSelect).toHaveBeenCalled();
	});
});

describe('SocialIconsList', () => {
	const allLocales = config.locales;
	allLocales.forEach(localeCode => {
		it(`renders for ${localeCode} locale`, () => {
			const socialIconList = shallow(<SocialIconsList localeCode={localeCode} />);
			expect(socialIconList).toMatchSnapshot();
		});
	});
});
