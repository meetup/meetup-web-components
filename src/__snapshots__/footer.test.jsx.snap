// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer exists 1`] = `
<footer
  className="column-item column-item--shrink"
  id="mupFooter"
>
  <Stripe
    className="footerStripe-main"
    inverted={true}
  >
    <Bounds
      className="bounds--footer"
      narrow={true}
    >
      <Section
        hasSeparator={true}
      >
        <Chunk>
          <a
            className="link link--white"
            href="/create/"
          >
            Create a Meetup
          </a>
        </Chunk>
      </Section>
      <Section
        className="border--none"
        hasSeparator={true}
      >
        <Flex
          direction="column"
          switchDirection="large"
        >
          <FlexItem
            className="margin--bottom"
          >
            <Flex
              direction="column"
              switchDirection="medium"
            >
              <FlexItem
                key="linkSet-0"
              >
                <FooterCategory
                  header="Your Account"
                  items={
                    Array [
                      <a
                        href="/register"
                    >
                        Settings
                    </a>,
                      <a
                        href="/logout"
                    >
                        Log out
                    </a>,
                      <a
                        href="/help"
                    >
                        Help
                    </a>,
                    ]
                  }
                />
              </FlexItem>
              <FlexItem
                key="linkSet-1"
              >
                <FooterCategory
                  header="Discover"
                  items={
                    Array [
                      <a
                        href="/groups"
                    >
                        Groups
                    </a>,
                      <a
                        href="/calendar"
                    >
                        Calendar
                    </a>,
                      <a
                        href="/topics"
                    >
                        Topics
                    </a>,
                      <a
                        href="/cities"
                    >
                        Cities
                    </a>,
                    ]
                  }
                />
              </FlexItem>
              <FlexItem
                key="linkSet-2"
              >
                <FooterCategory
                  header="Meetup"
                  items={
                    Array [
                      <a
                        href="/about"
                    >
                        About
                    </a>,
                      <a
                        href="/pro"
                    >
                        Meetup Pro
                    </a>,
                      <a
                        href="/jobs"
                    >
                        Careers
                    </a>,
                      <a
                        href="/apps"
                    >
                        Apps
                    </a>,
                      <a
                        href="/meetup_api"
                    >
                        API
                    </a>,
                    ]
                  }
                />
              </FlexItem>
            </Flex>
          </FlexItem>
          <FlexItem
            shrink={true}
          >
            <Flex
              direction="column"
              noGutters={true}
            >
              <FlexItem
                shrink={true}
              >
                <Chunk
                  className="align--center atMedium_align--left"
                >
                  <h4
                    className="text--bold margin--bottom"
                  >
                    Follow Us
                  </h4>
                  <SocialIconsList
                    localeCode="en-US"
                  />
                </Chunk>
              </FlexItem>
              <FlexItem>
                <Flex
                  direction="column"
                  justify={null}
                >
                  <FlexItem
                    className="margin--top"
                    shrink={true}
                  >
                    <Chunk>
                      <LanguageSelectInput
                        currentLocaleCode="en-US"
                        onChange={[Function]}
                      />
                    </Chunk>
                  </FlexItem>
                  <FlexItem
                    shrink={true}
                  >
                    <Chunk
                      className="align--center atMedium_align--left margin--top"
                    >
                      <AppBadges
                        getAppStorePhoto={[Function]}
                        language="en"
                      />
                    </Chunk>
                  </FlexItem>
                </Flex>
              </FlexItem>
            </Flex>
          </FlexItem>
        </Flex>
      </Section>
    </Bounds>
  </Stripe>
  <Stripe
    className="footerStripe-legal border--none"
    inverted={true}
  >
    <Bounds>
      <div
        className="padding--all"
      >
        <InlineBlockList
          className="text--small align--center atMedium_align--left footerList-legal"
          items={
            Array [
              <span>
                © Meetup 2018
            </span>,
              <span>
                Mock legal copy
            </span>,
            ]
          }
          separator="·"
        />
        <InlineBlockList
          className="text--small align--center atMedium_align--left margin--top"
          items={
            Array [
              <a
                href="/terms"
            >
                Terms
            </a>,
              <a
                href="/privacy"
            >
                Privacy
            </a>,
              <a
                href="/cookie_policy"
            >
                Cookies
            </a>,
            ]
          }
          separator="·"
        />
      </div>
    </Bounds>
  </Stripe>
</footer>
`;

exports[`Footer should render FooterCategory with linkSet 1`] = `
<Chunk>
  <h4
    className="text--bold margin--bottom"
  >
    Your Account
  </h4>
  <GridList
    className="text--secondary text--small"
    columns={
      Object {
        "default": 2,
        "medium": 1,
      }
    }
    itemClassNames="footer-item"
    items={
      Array [
        <a
          href="/register"
      >
          Settings
      </a>,
        <a
          href="/logout"
      >
          Log out
      </a>,
        <a
          href="/help"
      >
          Help
      </a>,
      ]
    }
  />
</Chunk>
`;

exports[`LanguageSelectInput exists 1`] = `
<WithErrorList(SelectInput)
  label="Language"
  labelClassName="visibility--a11yHide"
  name="languagePicker"
  options={
    Array [
      Object {
        "label": "English",
        "value": "en-US",
      },
      Object {
        "label": "Deutsch",
        "value": "de-DE",
      },
      Object {
        "label": "English (Australia)",
        "value": "en-AU",
      },
      Object {
        "label": "Español",
        "value": "es",
      },
      Object {
        "label": "Español (España)",
        "value": "es-ES",
      },
      Object {
        "label": "Français",
        "value": "fr-FR",
      },
      Object {
        "label": "Italiano",
        "value": "it-IT",
      },
      Object {
        "label": "日本語",
        "value": "ja-JP",
      },
      Object {
        "label": "한국어",
        "value": "ko-KR",
      },
      Object {
        "label": "Nederlands",
        "value": "nl-NL",
      },
      Object {
        "label": "Português",
        "value": "pt-BR",
      },
      Object {
        "label": "Polski",
        "value": "pl-PL",
      },
      Object {
        "label": "Русский",
        "value": "ru-RU",
      },
      Object {
        "label": "ไทย",
        "value": "th-TH",
      },
      Object {
        "label": "Türkçe",
        "value": "tr-TR",
      },
    ]
  }
/>
`;

exports[`SocialIconsList renders for de-DE locale 1`] = `
<InlineBlockList
  items={
    Array [
      <a
        href="https://www.facebook.com/meetupDE/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/MeetupDE/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetupde/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for en-AU locale 1`] = `
<InlineBlockList
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/Meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for en-US locale 1`] = `
<InlineBlockList
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/Meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for es locale 1`] = `
<InlineBlockList
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/MeetupES/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for es-ES locale 1`] = `
<InlineBlockList
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/MeetupES/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for fr-FR locale 1`] = `
<InlineBlockList
  items={
    Array [
      <a
        href="https://www.facebook.com/MeetupFR/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/MeetupFR/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/meetupfr/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for it-IT locale 1`] = `
<InlineBlockList
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/MeetupIT/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for ja-JP locale 1`] = `
<InlineBlockList
  items={
    Array [
      <a
        href="https://www.facebook.com/meetupjp/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/MeetupJP/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for ko-KR locale 1`] = `
<InlineBlockList
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/Meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for nl-NL locale 1`] = `
<InlineBlockList
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/Meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for pl-PL locale 1`] = `
<InlineBlockList
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/MeetupPL/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for pt-BR locale 1`] = `
<InlineBlockList
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/MeetupBR/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for ru-RU locale 1`] = `
<InlineBlockList
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/Meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for th-TH locale 1`] = `
<InlineBlockList
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/Meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;

exports[`SocialIconsList renders for tr-TR locale 1`] = `
<InlineBlockList
  items={
    Array [
      <a
        href="https://www.facebook.com/meetup/"
    >
        <Icon
            shape="external-facebookboxed"
            size="s"
        />
    </a>,
      <a
        href="https://twitter.com/Meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-twitter"
            size="s"
        />
    </a>,
      <a
        href="https://www.youtube.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-youtube"
            size="s"
        />
    </a>,
      <a
        href="https://www.instagram.com/meetup/"
    >
        <Icon
            className="margin--left"
            shape="external-instagram"
            size="s"
        />
    </a>,
      <a
        href="https://medium.com/meetup"
    >
        <Icon
            className="margin--left"
            shape="external-medium"
            size="s"
        />
    </a>,
    ]
  }
/>
`;
